firebase-dynamic-links/ktx/src/main/kotlin/com/google/firebase/dynamiclinks/ktx/FirebaseDynamicLinks.kt

```diff
@@ -0,0 +1,153 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.dynamiclinks.ktx
+
+import androidx.annotation.Keep
+import com.google.android.gms.tasks.Task
+import com.google.firebase.FirebaseApp
+import com.google.firebase.dynamiclinks.FirebaseDynamicLinks
+import com.google.firebase.components.Component
+import com.google.firebase.components.ComponentRegistrar
+import com.google.firebase.dynamiclinks.DynamicLink
+import com.google.firebase.dynamiclinks.ShortDynamicLink
+
+import com.google.firebase.ktx.Firebase
+import com.google.firebase.platforminfo.LibraryVersionComponent
+
+/** Returns the [FirebaseDynamicLinks] instance of the default [FirebaseApp]. */
+val Firebase.dynamicLinks: FirebaseDynamicLinks
+    get() = FirebaseDynamicLinks.getInstance()
+
+/** Returns the [FirebaseDynamicLinks] instance of a given [FirebaseApp]. */
+fun Firebase.dynamicLinks(app: FirebaseApp): FirebaseDynamicLinks {
+    return FirebaseDynamicLinks.getInstance(app)
+}
+
+/** Returns a [DynamicLink.AndroidParameters] object initialized using the [init] function. */
+fun androidParameters(
```

Comment:
Just a consideration:

If I understand it correctly the usage pattern for it will be along the lines of:

```kotlin
Firebase.dynamicLinks.createDynamicLink()
  .setDomainUriPrefix("domain")
  .setAndroidParameters(androidParameters {
    setMinimumVersion(16)
  }).build()
```

I think it can be improved a bit to read like:

```kotlin
Firebase.dynamicLinks.createDynamicLink {
  domainUriPrefix = "domain"
  androidParameters {
    minumumVersion = 16
  }
  iosParameters {
  }
}
```

While they look similar, the latter one is more concise, even though it would require to change all your builder functions to become extension methods of `DynamicLink.Builder` and add getters to all builders to make property syntax work. wdyt?

Reply:
Sounds Great to Me! :)